backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/predicate/impl/ElasticsearchNestedPredicateBuilder.java

```diff
@@ -35,6 +35,8 @@ public void nested(ElasticsearchSearchPredicateBuilder nestedBuilder) {
 	@Override
 	protected JsonObject doBuild(ElasticsearchSearchPredicateContext context,
 			JsonObject outerObject, JsonObject innerObject) {
+		context.explicitNested();
```

Comment:
I think we took another path in `LuceneSearchPredicateContext`: the context is immutable, and we create another one where needed. Which seems safer to me.

Here you need to forward some internal state too, so you could have something like this instead:

```
ElasticsearchSearchPredicateContext childContext = context.nest( absoluteFieldPath ); // Returns a new context
```

And then below:

```
QUERY_ACCESSOR.set( innerObject, nestedBuilder.build( childContext ) ); 
```